src/parcsr_ls/par_mgr.c

```diff
@@ -3573,15 +3588,35 @@ hypre_MGRSetGlobalSmootherAtLevel( HYPRE_Int     level,
                                                    HYPRE_MEMORY_HOST);
    }
 
-   /* Check if the requested level makes sense */
-   if (level < 0 || level >= max_num_coarse_levels)
+   /* Allocate level_smooth_type if needed */
+   if (!(mgr_data -> level_smooth_type))
    {
-      hypre_error_in_arg(2);
-      return hypre_error_flag;
+      (mgr_data -> level_smooth_type) = hypre_CTAlloc(HYPRE_Int,
+                                                      max_num_coarse_levels,
+                                                      HYPRE_MEMORY_HOST);
    }
 
    (mgr_data -> level_smoother)[level] = smoother;
 
+   /* Obtain corresponding smoother type */
```

Comment:
See previous comment above about doing this check once. Level_smooth_type should be 'unset' for this level (and in the previous call) since the mgr_solve routine checks both level_smooth_type  (first) and then level_smoother to call the smoother. This smoother set here may not be called then.

Reply:
Resolved offline